### PR TITLE
refactor: use quota.Equals consistently for resourceQuota compare

### PR DESCRIPTION
**Pull request by pandaamanda**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:

uniformly use quota.Equals like bellow and remove dep k8s.io/apimachinery/pkg/api/equality.

https://github.com/kubernetes/kubernetes/blob/e234917e0a3e4bb384ef14c198742643a7e170fd/pkg/controller/resourcequota/resource_quota_controller.go#L135
https://github.com/kubernetes/kubernetes/blob/e234917e0a3e4bb384ef14c198742643a7e170fd/pkg/controller/resourcequota/resource_quota_controller.go#L364

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: pkg/controller/resourcequota/resource_quota_controller.go [hunk 1]
```diff
@@ -43,6 +40,7 @@ import (
 	"k8s.io/client-go/tools/cache"
 	"k8s.io/client-go/util/workqueue"
 	"k8s.io/controller-manager/pkg/informerfactory"
+	"k8s.io/klog/v2"
```
Comment (yangjunmyfm192085) [inline pkg/controller/resourcequota/resource_quota_controller.go:43]: Why do we need to modify here?

Comment (pandaamanda) [reply pkg/controller/resourcequota/resource_quota_controller.go:43]: Just a layout style, i think it's better.

**Comment by pandaamanda** [comment]
kindly ping for review
/assign @cheftako

**Pull request closed without merging**